### ISSUE TITLE
Emoji shortcodes should include skin color

### ISSUE THREAD

**Issue by SoniEx2** [tootsuite/mastodon#14724]
### Pitch

When typing emoji shortcodes, it should be possible to prefix them with a skin color to find emoji with a skin color. Obviously showing them on the emoji picker would make it unusable because of having too many of the same emoji (unless padded/wrapped to make it easier), so maybe keep them just for the shortcodes.

### Motivation

Antiracism. You should be able to select skin color on the web interface, particularly on desktop. There are workarounds with IMEs and stuff but ugh linux IMEs are not great about it either, and I also don't think one should need to install and configure an IME for this. :/

This benefits ppl who a) don't have the technical ability to install and configure an IME, b) may not be allowed to install and configure an IME (such as in work environments) or c) can't install and configure an IME for any other reason not mentioned above, and are using mastodon on a desktop computer.

**Comment by SoniEx2** [comment]
Related, but different: #5498

**Comment by marrus-sh** [comment]
a typical approach to this is to simply assign `:skin-tone-1:`, `:skin-tone-2:`, `:skin-tone-3:`, `:skin-tone-4:`, `:skin-tone-5:` to the skin tone modifiers and then use e.g., `:ok_hand::skin-tone-3:` to insert the OK hand and the skin tone in sequence

this is what Discord does, for example

note this is a *suffix*, not prefix, because that’s how Unicode does things

**Comment by SoniEx2** [comment]
we thought you needed to use ZWJs with those?

**Comment by marrus-sh** [comment]
nope, skin tones are modifiers and function essentially like combining characters which combine with the previous emoji character

**Comment by SoniEx2** [comment]
oh, okay.

**Comment by ThibG** [comment]
> (**edit:** does mastodon still not allow adjacent colons when stringing together shortcodes?? because uh, you’ll have to fix that for this solution to work.)

Mastodon doesn't perform shortcode → emoji conversion on the backend anymore, it only does the conversion on-the-fly with the emoji picker, so you'd have the actual unicode emoji and not the `:shortcode:` when you're finished inputting the first emoji.

However, the picker doesn't show up when `:` is typed with some non-blank characters directly preceding it.
